### PR TITLE
preUpdate Event Listener On Uploaded Imagery

### PR DESCRIPTION
**Pull request by matt9mg**
I've noticed a scenario (a bit of an edge case when uploading imagery).

If you have an entity (say a brochure) but you can give it a title and an image and the image title can be updated without the image being updated (the file type is not an UploadedFile but a File on save it'll persist the full file path instead of the desired filename. The above is a safety measure to ensure the above.

### CODE REVIEW

File: controller/upload_file.rst [hunk 1]
```diff
@@ -362,6 +363,10 @@ automatically upload the file when persisting the entity::
             if ($file instanceof UploadedFile) {
                 $fileName = $this->uploader->upload($file);
                 $entity->setBrochure($fileName);
+            } elseif($file instanceof File) {
```
Comment (Simperfit) [inline controller/upload_file.rst]: it should be : 
`elseif ($file instanceof File) {`

Comment (matt9mg) [reply controller/upload_file.rst]: Thank you @Simperfit updated!

File: controller/upload_file.rst [hunk 2]
```diff
@@ -362,6 +363,10 @@ automatically upload the file when persisting the entity::
             if ($file instanceof UploadedFile) {
                 $fileName = $this->uploader->upload($file);
                 $entity->setBrochure($fileName);
+            } elseif ($file instanceof File) {
+                // prevents the full file path being saved on updates
+                // as the path is set on the postLoad listener
+                $entity->setBrocure($file->getFilename());
```
Comment (javiereguiluz) [inline controller/upload_file.rst]: Typo: `setBrocure` -> `setBrochure`

Comment (matt9mg) [reply controller/upload_file.rst]: @javiereguiluz thank you for the feedback :)

**Review by rzilin — APPROVED** [review]

**Comment by matt9mg** [comment]
*cough* 😊

**Comment by javiereguiluz** [comment]
@matt9mg sorry for not having reviewed this sooner. Let's ping @vudaltsov because he recently reviewed another change proposal related to file upload. Thanks!

**Review by HeahDude — APPROVED** [review]
Thanks for this PR!

**Comment by xabbuh** [comment]
Thank you @matt9mg.

**Pull request closed without merging**